matlab/src/cpp/arrow/matlab/array/proxy/timestamp_array.cc

```diff
@@ -0,0 +1,92 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/matlab/array/proxy/timestamp_array.h"
+
+#include "arrow/matlab/error/error.h"
+#include "arrow/matlab/bit/pack.h"
+#include "arrow/matlab/bit/unpack.h"
+
+#include "arrow/matlab/type/time_unit.h"
+#include "arrow/util/utf8.h"
+#include "arrow/type.h"
+#include "arrow/builder.h"
+
+
+namespace arrow::matlab::array::proxy {
+
+    namespace {
+        const uint8_t* getUnpackedValidityBitmap(const ::matlab::data::TypedArray<bool>& valid_elements) {
```

Comment:
Can we unify this and a similar function in `numeric_array.h`?

Reply:
Definitely! That's a good idea.

Reply:
Actually, it looks like @kevingurney has already done this. He's currently working on adding an `arrow.array.StringArray` class and created a shared utility called [unpacked_as_ptr](https://github.com/mathworks/arrow/blob/cd4c7cc36e2d21894399387fb51ceda9cf86d532/matlab/src/cpp/arrow/matlab/bit/unpack.cc#L42).  Is it alright if I create an issue to use that utility in `timestamp_array.cc` once Kevin's changes are merged?

Reply:
OK. I merge this.

Reply:
Thanks @kou! I've made an [issue](https://github.com/apache/arrow/issues/36335) to track this.